src/languages/php.js

```diff
@@ -60,7 +60,13 @@ export default function(hljs) {
       HEREDOC
     ]
   };
-  const NUMBER = {variants: [hljs.BINARY_NUMBER_MODE, hljs.C_NUMBER_MODE]};
+  const NUMBER = {
+    className: 'number',
+    variants: [
+      { begin: `\\b0b[01]+(?:_[01]+)*\\b|\\b0o[0-7]+(?:_[0-7]+)*\\b|\\b0x[\\da-f]+(?:_[\\da-f]+)*\\b|(?:\\b\\d+(?:_\\d+)*(\\.(?:\\d+(?:_\\d+)*))?|\\B\\.\\d+)(?:e[+-]?\\d+)?` },
```

Comment:
Nice.

Please expand this into multiple variants for each variant of number vs a huge regex that tends to be impossible to read and maintain over time.

Reply:
Thank you. It's now expanded into four variants - binary, octal, hex, and decimal.